lib/index.js

```diff
@@ -796,23 +870,26 @@ function initDB(peer, config) {
   /**
    * @param {Tangle} accountTangle
    * @param {KeypairPublicSlice} keypair
-   * @returns {Set<AccountPower>}
    */
-  function getAccountPowers(accountTangle, keypair) {
+  function getAccountPowers(accountTangle, keypair, cb) {
     const powers = new Set()
     for (const msgID of accountTangle.topoSort()) {
-      const msg = get(msgID)
-      if (!msg?.data) continue
-      /** @type {AccountData} */
-      const data = msg.data
-      if (data.action !== 'add') continue
-      if (data.key.algorithm !== keypair.curve) continue
-      if (data.key.bytes !== keypair.public) continue
-      if (data.powers) {
-        for (const power of data.powers) {
-          powers.add(power)
+      // TODO: oops fix for loop
```

Comment:
in the middle of fixing this but getting annoying/hard to debug test failures

Reply:
re-enabled typechecking to hopefully find a type error around this but didn't seem to help really